ideascube/configuration/models.py

```diff
@@ -0,0 +1,76 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+from .errors import (
+    InvalidConfigurationValue,
+    InvalidStoredConfigurationValue,
+    NoSuchConfigurationKey,
+    NoSuchConfigurationNamespace,
+    )
+from .registry import REGISTRY
+
+
+class Configuration(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @staticmethod
+    def _get_key_registry(namespace, key):
+        try:
+            namespace_registry = REGISTRY[namespace]
+
+        except KeyError:
+            raise NoSuchConfigurationNamespace(namespace)
+
+        try:
+            return namespace_registry[key]
+
+        except KeyError:
+            raise NoSuchConfigurationKey(namespace, key)
+
+    @classmethod
+    def get(cls, namespace, key):
+        key_registry = cls._get_key_registry(namespace, key)
+        expected_type = key_registry['python_type']
+
+        config = cls.objects.get(namespace=namespace, key=key)
+
+        if not isinstance(config.value, expected_type):
```

Comment:
> I still disagree to return the default value if a value is present but with a wrong type.

Me too, but I gave up on that and the next push will do as @yohanboniface asked.

> it even allow to handle wrong type value stored (int(5) == int('5')).

And this would be a completely invalid value.